nvtabular/loader/torch.py

```diff
@@ -174,8 +180,48 @@ def _get_sparse_tensor(self, values, indices, num_rows, seq_limit):
             sparse_tensor = sparse_tensor.to_dense()
         return sparse_tensor
 
-    def _build_sparse_tensor(self, values, offsets, diff_offsets, num_rows, seq_limit):
+    def _build_sparse_tensor_helper_process_column(self, col: torch.Tensor) -> torch.Tensor:
+        """Process column by increasing blocks for use in left padding."""
+        col = col.tolist()
+        prev, curr = 0, 0
+        while curr < len(col):
```

Comment:
this is the while loop I am talking about @lesnikow

Reply:
This is good to know, thank you. I have not implemented anything optimized over this yet. I would like to hear your feedback first on the `tensorflow` side. This torch implementation is also operating on `torch.sparse` tensors, where `torch.functional.nn.pad` and `torch.flip` are not implemented on `torch.sparse` tensors. Would you have any guidance on how to proceed for this `torch.sparse` case?